internal/statemachine/classic_states.go

```diff
@@ -1255,6 +1259,29 @@ func (stateMachine *StateMachine) populateClassicRootfsContents() error {
 	return nil
 }
 
+// Set a default locale if one is not configured beforehand by other customizations
+func (stateMachine *StateMachine) setDefaultLocale() error {
+	classicStateMachine := stateMachine.parent.(*ClassicStateMachine)
+
+	defaultPath := filepath.Join(classicStateMachine.tempDirs.chroot, "etc", "default")
+	localePath := filepath.Join(defaultPath, "locale")
+	localeBytes, err := osReadFile(localePath)
+	if err == nil && strings.Contains(string(localeBytes), "LANG=") {
+		return nil
+	}
+
+	err = osMkdirAll(defaultPath, 0755)
+	if err != nil {
+		return fmt.Errorf("Error creating default directory: %s", err.Error())
+	}
+
+	err = osWriteFile(localePath, []byte("# Default Ubuntu locale\nLANG=C.UTF-8\n"), 0644)
```

Comment:
This overwrites any previous file right? In practice this is extremely likely to be fine but there are ways it could incorrect I suppose, like if the image creator wants to set LC_MESSAGES to a different value to LC_COLLATE for some insane reason.